app/src/main/java/com/team41/boromi/dbs/UserDB.java

```diff
@@ -102,4 +101,31 @@ public User pushUser(User user) {
       return null;
     }
   }
+
+  /**
+   * async gets result whether if username is unique or not.
+   * @param username
+   * @return
+   */
+
+  public Boolean isUsernameUnique(String username){
+    QuerySnapshot res;
+
+    // Queries firestore for the username
+    try {
+      res = Tasks.await(
+              usersRef.whereEqualTo("username", username).get(),
+              DB_TIMEOUT,
+              TimeUnit.MILLISECONDS
+      );
+    } catch (Exception e) { // Request failed
+      Log.w(TAG, e.getCause());
+      return null;
+    }
+    if (res.isEmpty()) { // No user found
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
Can be simplified to
'''java
return res.isEmpty()
'''

Though I guess that's just preference

Reply:
yea true thanks!

Reply:
fixed c3eca5d